### ISSUE TITLE
MD5 hash field on authorize.net and opencart aim module are not compatible. 

### ISSUE THREAD

**Issue by oasisfleeting** [opencart/opencart#1419]
Ultimately, the true fix for the authorize.net aim module would be to limit the md5 hash field to 20 characters so that it matches the hash entered in to authorize.net

====== Steps to reproduce ======

Using a phrase will work, but that isn't the REAL problem.
The root of the problem is this:
1.) Grab a hash from any site like http://www.adamek.biz/md5-generator.php , an MD5 hash is always 32 characters
2.) Enter the hash into Authorize.net's md5 hash and compare that to the hash you created. You'll notice that on Authorize.net's MD5 hash the field length is limited to 20 characters but the md5 hash field in opencart is not. This is the causing invalid hash collisions.

Solution: Create your hash from a password or for example i_love_lucy.
i_love_lucy will hash into bcd91b9012a23f973cd8baca7eeeb5de but only 20 of those 32 characters will be allowed into the authorize.net md5 hash field.
So in conclusion the hash that needs to be put into opencart auth.net aim is bcd91b9012a23f973cd8

Do not comment out the hash checking code as this will more than likely compromise your shops security.

Ultimately, the true fix for the authorize.net aim module would be to limit the md5 hash field to 20 characters so that it matches the hash entered in to authorize.net

Opencart 1.5.6.4

**Comment by mrwconnected** [comment]
We've done this (used the shorter hash) in OpenCart but still get errors on some transactions (MD5 invalid error).  Anyone found a better solution?
